src/HealthChecks.Rabbitmq/DependencyInjection/RabbitMQHealthCheckBuilderExtensions.cs

```diff
@@ -139,12 +140,9 @@ public static class RabbitMQHealthCheckBuilderExtensions
         IEnumerable<string>? tags = default,
         TimeSpan? timeout = default)
     {
-        builder.Services
-            .AddSingleton(sp => new RabbitMQHealthCheck(connectionFactory(sp)));
-
         return builder.Add(new HealthCheckRegistration(
             name ?? NAME,
-            sp => sp.GetRequiredService<RabbitMQHealthCheck>(),
+            sp => new RabbitMQHealthCheck(connectionFactory(sp)),
```

Comment:
You changed creation semantics here. Before `IConnection` instance was created only once. Now it's created (I understand that user delegate does not necessarily _create_ it but I suppose it's so the majority of time) each call to `HealthCheckRegistration.Factory`. At least I would add xml comment that `connectionFactory` delegate is called each time. The same for other methods accepting delegates.